examples/goldilocks_models/initial_guess.cc

```diff
@@ -0,0 +1,287 @@
+  VectorXd current_gamma = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+      + string("_task.csv"));
+  int iter_start;
+  string data_dir;
+  //if use solutions from database
+  if(use_database && (iter<2)){
+    if(iter==0){
+      //use nominal database
+      data_dir = "../dairlib_data/goldilocks_models/database/robot_" +
+          to_string(robot) + "_nominal/";
+    }
+    else{
+      //use initial database
+      data_dir = "../dairlib_data/goldilocks_models/database/robot_" +
+          to_string(robot) + "_initial/";
+    }
+    //use solutions in database to calculate interpolated initial guess
+    VectorXd initial_guess;
+    //take out corresponding w and calculate the weight for interpolation
+    MatrixXd w_gamma;
+    VectorXd weight_gamma;
+
+    //calculate the weighted sum of past solutions
+    int sample_num = 0;
+    while(file_exist(data_dir + to_string(sample_num)+string("_0_is_success.csv"))){
+      //check if this sample is success
+      int is_success = (readCSV(data_dir + to_string(sample_num)
+                                    + string("_0_is_success.csv")))(0,0);
+      if(is_success == 1) {
+        //extract past gamma
+        VectorXd past_gamma = readCSV(data_dir + to_string(sample_num)
+                                          + string("_0_gamma.csv"));
+        //calculate the weight for each sample using the third power of the difference between gamma
+        VectorXd dif_gamma = (past_gamma - current_gamma).array().abs()*gamma_scale.array();
+        VectorXd dif_gamma2 = dif_gamma.array().pow(2);
+        double distance_gamma =  (dif_gamma.transpose() * dif_gamma2)(0,0);
+//        //extract the solution of this sample
+        VectorXd w_to_interpolate = readCSV(data_dir + to_string(sample_num)
+                                                + string("_0_w.csv"));
+        //concatenate the weight and solution for further calculation
+        w_gamma.conservativeResize(w_to_interpolate.rows(),w_gamma.cols()+1);
+        w_gamma.col(w_gamma.cols()-1) = w_to_interpolate;
+        weight_gamma.conservativeResize(weight_gamma.rows()+1);
+        weight_gamma(weight_gamma.rows()-1)= 1 /distance_gamma;
+      }
+      sample_num = sample_num+1;
+    }
+    DRAKE_DEMAND(weight_gamma.rows()>0);
+    //    normalize weight
+    weight_gamma = weight_gamma / weight_gamma.sum();
+    initial_guess = w_gamma * weight_gamma;
+    //    save initial guess and set init file
+    initial_file_name = to_string(iter) + "_" + to_string(sample)
+        + string("_initial_guess.csv");
+    writeCSV(directory + initial_file_name, initial_guess);
+
+  }
+  else {
+    //    For iter = 1, use the solutions from iteration 0 to set the initial guess
+    if (iter == 1) {
+      iter_start = 0;
+    }
+      //if iter > 1, use the results from past iterations to interpolate and set the initial guess
+    else {
+      iter_start = 1;
+    }
```

Comment:
For the situation of getting nominal cost(iter = 0), only when flag `is_use_database=true`, the program will enter here and it will be handled at the beginning of this function.

Reply:
Got it. I think catching the "iter==0" case and throwing an error here would be a good idea for preventing future misuse of `setInitialGuessByInterpolation`